pkg/vsphere/tasks/waiter.go

```diff
@@ -268,6 +268,23 @@ func IsNotFoundError(err error) bool {
 	return false
 }
 
+func IsInvalidPowerStateError(err error) bool {
```

Comment:
This function is likely overlycomplicated. But, I do not see a reason not to cover all the bases here. Feel free leave additional thoughts!